src/main/java/http_server/RequestParser.java

```diff
@@ -0,0 +1,82 @@
+package http_server;
+
+import java.util.HashMap;
+
+public class RequestParser {
+    HashMap<String, String> requestCollection = new HashMap<>();
+    HashMap<String, String> headersCollection = new HashMap<>();
+
+    public RequestParser(String request) {
+        String[] splitRequest = request.trim().split("\r\n");
+
+        responseLineParser(splitRequest);
+        bodyParser(splitRequest);
+        headerParser(splitRequest);
+    }
+
+    public String method() {
+        return requestCollection.get("method");
+    }
+
+    public String body() {
+        return requestCollection.get("responseBody");
+    }
+
+    public String route() {
+        return requestCollection.get("route");
+    }
+
+    public HashMap<String, String> headers() {
+        return headersCollection;
+    }
+
+    private void responseLineParser(String[] splitRequest) {
```

Comment:
Why is there a method called `responseLineParser` in the `RequestParser` class>

Reply:
Misnamed! Should be `requestLineParser`